### PR TITLE
Build and test for Desktop framework in xplat

### PR DESCRIPTION
**Pull request by rainersigwald**
This builds on #842--comments on `cibuild.cmd` should go there.

On my machine, I get a mysterious failure loading `System.Runtime` _only_ in the Engine tests. I haven't figured out why yet. Will be interesting to see if the CI builds produce the same thing.

### CODE REVIEW

**Comment by rainersigwald** [comment]
@dotnet-bot test OSX Build for CoreCLR please

File: cibuild.cmd [hunk 1]
```diff
@@ -1,5 +1,148 @@
...
+echo   --scope ^<scope^>                Scope of the build ^(Compile / Test^)
+echo   --target ^<target^>              CoreCLR or Desktop ^(default: Desktop^)
+echo   --host ^<host^>                  CoreCLR or Desktop ^(default: Desktop^)
+echo   --skip-bootstrap               Do not rebuild msbuild with local binaries
+exit /b 1
+
+:error
+echo.
+echo ---------------------------------------
+echo - RebuildWithLocalMSBuild.cmd FAILED. -
```
Comment (AndyGerlicher) [inline cibuild.cmd]: >RebuildWithLocalMSBuild [](start = 7, length = 23)

cibuild.cmd FAILED?

**Comment by AndyGerlicher** [comment]
:shipit:

File: cibuild.cmd (hunk 1 above)
Comment (rainersigwald) [inline cibuild.cmd]: Good catch. Fixed in #842.

**Comment by rainersigwald** [comment]
I'd like to go in with this even though I haven't fixed the `System.Runtime` thing. It doesn't make the red CI any worse and raises the bar . . .

@dotnet-bot test this please

**Comment by rainersigwald** [comment]
@dotnet-bot test OSX Build for CoreCLR please

File: src/Shared/NativeMethodsShared.cs [hunk 2]
```diff
@@ -613,17 +613,42 @@ internal static string FrameworkBasePath
...
+                            _systemInformationInitialized = true;
+                        }
+                    }
+                }
+                return _systemInformation;
+            }
+        }
+
+        private static SystemInformationData _systemInformation;
+        private static bool _systemInformationInitialized;
```
Comment (cdmihai) [inline src/Shared/NativeMethodsShared.cs]: Make this volatile so threads don't work on cached values?

Comment (rainersigwald) [inline src/Shared/NativeMethodsShared.cs]: [Allegedly](http://stackoverflow.com/a/1964832) unnecessary--if two threads get stale cached values, they'll both attempt to acquire the lock, one will succeed and pass the inner check, the other will block, then get the lock and fail the inner check.

Comment (cdmihai) [inline src/Shared/NativeMethodsShared.cs]: OK, sold. Good enough for CLR :)

**Comment by cdmihai** [comment]
:shipit:

**Pull request merged**